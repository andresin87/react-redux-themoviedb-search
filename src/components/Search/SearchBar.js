import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Input } from 'antd';

import './SearchBar.css';
import { fetchSearch } from '../../actions';

const Search = Input.Search;

class SearchBar extends PureComponent {
  render() {
    const { fetchSearch } = this.props;
    return (
      <div className="SearchBar">
        <Search
          placeholder="Search for a film..."
          enterButton="Search"
          size="large"
          onSearch={value => fetchSearch(value)}
        />
      </div>
    );
  }
}

SearchBar.propTypes = {
  fetchSearch: PropTypes.func.isRequired,
};

export default connect(
  null,
  { fetchSearch }
)(SearchBar);
