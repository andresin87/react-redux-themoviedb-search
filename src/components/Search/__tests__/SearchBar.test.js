import React from 'react';
import ReactDOM from 'react-dom';
import TestUtils from 'react-dom/test-utils';
import renderer from 'react-test-renderer';
import { Input } from 'antd';
import { Provider } from 'react-redux';

import SearchBar from '../SearchBar';
import store from '../../../config/mockStore';
import { SEARCH_FETCH_REQUEST } from '../../../actions';

describe('SearchBar component', () => {
  const component = (
    <Provider store={store}>
      <SearchBar />
    </Provider>
  );

  it('Renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(component, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('Snapshot matchs', () => {
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Check that the search button launchs an action', () => {
    const tree = TestUtils.renderIntoDocument(component);
    const search = TestUtils.findRenderedComponentWithType(tree, Input.Search);
    const { onSearch } = search.props;

    //Call the function
    const query = 'Fight club';
    onSearch(query);

    const expectedActions = [
      {
        type: SEARCH_FETCH_REQUEST,
        payload: {
          meta: {
            query,
          },
        },
      },
    ];
    expect(store.getActions()).toEqual(expectedActions);
  });
});
