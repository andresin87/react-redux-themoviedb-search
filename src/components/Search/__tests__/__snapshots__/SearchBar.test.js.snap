// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar component Snapshot matchs 1`] = `
<div
  className="SearchBar"
>
  <code
    name="Antd Search"
    props={
      Object {
        "enterButton": "Search",
        "onSearch": [Function],
        "placeholder": "Search for a film...",
        "size": "large",
      }
    }
  />
</div>
`;
